### PR TITLE
Added XGBoost regression model example

### PR DESCRIPTION
**Pull request by evsmithx**
Also added notes to the docs on how to make a learning algorithm work with collective learning. #126 should be merged first!

### CODE REVIEW

**Review by MissingNO57 — APPROVED** [review]
LGTM

File: examples/xgb_reg_boston.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        self.model = xgb.train(self.params, self.xg_train, self.n_steps_per_round)
+
+        self.model_file_base = f"/tmp/model_{self.worker_id}"
+        self.n_saves = 0
+        self.vote_score = self.test(self.xg_train)
+
+    def mli_propose_weights(self) -> Weights:
+        current_model = self.mli_get_current_weights()
+
+        # xgb_model param means training starts from previous weights
```
Comment (JuanBesa) [inline examples/xgb_reg_boston.py:60]: I would add a link to the docs discussion here

File: docs/intro_tutorial_mli.md [hunk 2]
```diff
@@ -27,6 +27,43 @@ There are four methods that need to be implemented:
     The old weighs of the model are discarded and replaced by the new weights.
 4. `current_weights` should return the current weights of the model.
 
+
+## Algorithms that work with colearn:
+These conditions need to be fulfilled for algorithms to work with collective learning:
+
+* The algorithm needs to have a "warm start", i.e. model fitting is incremental so that the previous model is
```
Comment (JuanBesa) [inline docs/intro_tutorial_mli.md:34]: I would turn this around. i.e. Model fitting must be incremental, i.e. the algorithm needs to have a "warm start"

**Review by JuanBesa — APPROVED** [review]
Looks good!

Two minor comments.

**Review by JuanBesa — APPROVED** [review]
Lgtm

**Pull request merged**